packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js

```diff
@@ -53,13 +53,22 @@ SelfUtils.parse = (self, deviceId) => {
       selfId: self.id,
       selfIdentity: SelfUtils.getSelfIdentity(self),
       selfUrl: self.url,
-      removed: self.removed
+      removed: self.removed,
+      roles: SelfUtils.getRoles(self)
     };
   }
 
   return null;
 };
 
+SelfUtils.getRoles = (self) => (self?.controls?.role?.roles || []).reduce((roles, role) => {
```

Comment:
Does changing it to a filter reduce the complexity? It still needs to be a filter plus a map because I want the type of the role, not just the role itself. e.g.
```js
SelfUtils.getRoles = (self) => (self?.controls?.role?.roles || []).filter((role) => role.hasRole).map((role) => role.type);
```

Reply:
My bad, I missed the mapping part. 
The reducer is better her.